### PR TITLE
Upgrade miekg/pkcs11 from v1.0.3 to v1.1.1

### PR DESCRIPTION
**Pull request by tstromberg**
Currently, the containers/ocicrypt library does not build on OpenBSD:

```
$ go test .
# github.com/miekg/pkcs11
ld: error: unable to find library -ldl
cc: error: linker command failed with exit code 1 (use -v to see invocation)
FAIL	github.com/containers/ocicrypt [build failed]
FAIL
```

This PR upgrades the miekg/pkcs11 component to a version that compiles cleanly on OpenBSD. Here are the functional PR's included in the miekg/pkcs11 upgrade from v1.0.3:

* https://github.com/miekg/pkcs11/pull/124
* https://github.com/miekg/pkcs11/pull/133
* https://github.com/miekg/pkcs11/pull/137
* https://github.com/miekg/pkcs11/pull/140
* https://github.com/miekg/pkcs11/pull/143
* https://github.com/miekg/pkcs11/pull/150

For a more specific diff , see https://github.com/miekg/pkcs11/compare/v1.0.3...v1.1.1

### CODE REVIEW

**Review by lumjjb — APPROVED** [review]
LGTM, would like to run this by @stefanberger first before merging since he is more familiar with the pkcs11 sections.

**Comment by stefanberger** [comment]
Hm, I hope it doesn't have any negative side effect on imgcrypt. I may try locally first, but it may take a while...

**Comment by stefanberger** [comment]
> For a more specific diff , see [miekg/pkcs11@v1.0.3...v1.1.1](https://github.com/miekg/pkcs11/compare/v1.0.3...v1.1.1)

I think we should be good...

**Comment by tstromberg** [comment]
Excellent. Please merge this once you get a chance. Thanks!

**Pull request merged**
